### PR TITLE
feat: fix(ogc): remediate protocol correctness gaps across maps, tiles, and wfs (#573)

### LINKED ISSUE #573: fix(ogc): remediate protocol correctness gaps across maps, tiles, and wfs
## Why

The audit identified a cluster of protocol-correctness issues across the OGC Maps, OGC Tiles, and WFS surfaces. They are all user-visible contract/compliance bugs, they share the same protocol-validation mindset, and they should be fixed together instead of opened as four separate tickets.

## Scope

In scope:
- ensure OGC Maps requests are classified as OGC so they receive OGC-specific error formatting
- transform OGC Tiles extents to CRS84 where required by the contract
- stop assuming little-endian WKB in the OGC Tiles renderer
- return WFS 2.0 XML ExceptionReport payloads where the endpoint contract advertises XML exceptions
- add or update targeted protocol tests for the above behaviors

Out of scope:
- broader OGC/WFS feature additions
- unrelated docs-only OGC spec drift already covered by #557
- unrelated audit findings from #571

## Acceptance Criteria

- `/ogc/maps` requests are classified into the OGC error/response path rather than generic Problem Details handling.
- OGC Tiles collection extents are emitted in CRS84-consistent form where the contract requires it.
- OGC Tiles WKB reading does not silently mis-handle big-endian payloads.
- WFS 2.0 error responses use XML ExceptionReport output in the advertised error path.
- Targeted tests cover the regression points for each corrected protocol behavior.

## Primary Repo

honua-server

## Repo Scope

honua-server

## Parent Audit Ticket

- Parent: #571

## Confirmed Findings

- `[correctness]` ProtocolRequestClassifier missing `/ogc/maps` (`src/Honua.Server/Features/Infrastructure/Models/ProtocolRequestClassifier.cs:25-28`)
- `[correctness]` OGC Tiles extent CRS inconsistency (`src/Honua.Server/Features/OgcTiles/CollectionsEndpoints.cs:229-248`)
- `[correctness]` OGC Tiles WKB byte-order assumption (`src/Honua.Server/Features/OgcTiles/TileRenderer.cs:88`)
- `[correctness]` WFS 2.0 error responses should be XML ExceptionReport (`src/Honua.Server/Features/Wfs20/Wfs20DispatcherEndpoint.cs`)

## Notes

This bundle is intentionally protocol-scoped so the fixes can share validation and review context instead of becoming four separate micro-tickets.

### PR DESCRIPTION
**Pull request by mikemcdougall**
# Pull Request

## Issue Link

Fixes #573
## Summary

- **Thread 2**: `TileRenderer.cs:89` reads byte-order flag; all reads use `BinaryPrimitives` helpers (lines 254-268)

Replied to both review threads with evidence pointing to the specific lines in the PR diff.
## Changes Made

- **Thread 2**: `TileRenderer.cs:89` reads byte-order flag; all reads use `BinaryPrimitives` helpers (lines 254-268)
- Replied to both review threads with evidence pointing to the specific lines in the PR diff.
## Testing

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Architecture tests pass
- [x] Local pre-PR validation passed (`scripts/pre-pr-check.sh`)
## Coverage Impact

- Line coverage: Not measured locally
- Branch coverage: Not measured locally
## Breaking Changes

None
## Additional Context

Decisions:
- Both review threads are false positives (reviewer analyzed base branch, not PR diff). Replied with evidence; no code changes required.

Follow-ons:
None blocking.

Code kaizen:
Skipped for docs/discovery delivery.
## Pre-PR Checklist (for contributor)
- [ ] Ran `scripts/pre-pr-check.sh` and all checks passed
- [ ] Commit message follows format: `type: description (#issue-number)`
- [ ] PR title matches main commit message
- [ ] Issue number linked above
- [ ] Tests added for new functionality
- [ ] Documentation updated if needed
- [ ] If protocol/auth behavior changed, updated MVP compatibility contract and release checklist notes
- [ ] If breaking admin/control-plane API changes: updated migration guide and versioning policy docs
- [ ] If breaking gRPC/proto wire changes: confirmed with explicit review and documented in migration guide

## Reviewer Checklist
- [ ] Code follows project architecture (vertical slices, no controllers)
- [ ] Tests cover happy path and edge cases
- [ ] No reflection in hot paths (AOT compatible)
- [ ] Dependency limits respected (max 5 per endpoint)
- [ ] Error handling follows project patterns
- [ ] Security considerations addressed

## Gate Impact

- [x] PR gates (build, test, governance)
- [ ] Nightly gates (conformance, performance, security)
- [ ] Release gates (packaging, publishing)
- [ ] Deploy gates (promotion, post-apply validation)
- [ ] None — no gate impact

## Docs or Contract Impact

- [ ] OpenAPI spec changed
- [ ] Protobuf/gRPC contract changed
- [ ] Control plane SDK surface changed
- [x] Documentation updated
- [ ] None — no docs or contract impact

## Release/Deploy Impact

- [ ] Requires coordinated release across repos
- [ ] Requires database migration
- [ ] Requires infrastructure changes
- [ ] Requires environment variable or secret changes
- [x] None — standard merge-and-release flow

### CODE REVIEW

File: docs/user/specifications/ogc-api-tiles-coverage.md [hunk 1]
```diff
@@ -15,6 +15,18 @@ This page summarizes Honua MVP support for OGC API Tiles.
 | Tile retrieval | Implemented | Vector/raster responses by collection/tileset/tile matrix |
 | OpenAPI (`/ogc/tiles/openapi.json`) | Implemented | OpenAPI description for tiles API |
 
+## Protocol Correctness Notes
+
+### Collection Spatial Extents (CRS84)
+
+The OGC Tiles spec requires collection spatial extents to use CRS84 coordinates. Honua transforms extents from the layer's native CRS to CRS84 using an in-memory transformer that supports WGS 84 (EPSG:4326) and WebMercator variants (EPSG:3857, 900913, 102100, 102113, 3785). This matches the pattern used by OGC API Features and WFS 2.0 collections.
+
+When a layer uses a CRS that cannot be reliably transformed in-memory, the spatial extent is **omitted** rather than emitted with non-CRS84 coordinates. Clients should handle absent spatial extents gracefully.
```
[the remark this answers is not in the document]
Comment (mikemcdougall) [reply docs/user/specifications/ogc-api-tiles-coverage.md:24]: This is addressed by commit `3d648047`. The PR diff shows `CollectionsEndpoints.cs:229-258` now calls `OgcExtentTransformer.TryTransformToCrs84()` and emits `Crs = OgcFeaturesUtilities.Crs84Uri`. Unsupported CRS omits the spatial extent entirely. The documented behavior matches the implementation.

File: docs/user/specifications/ogc-api-tiles-coverage.md [hunk 2]
```diff
@@ -15,6 +15,18 @@ This page summarizes Honua MVP support for OGC API Tiles.
...
+
+### WKB Byte Order
+
+The tile renderer reads WKB geometry payloads with endian awareness. Both little-endian (NDR, byte-order flag `0x01`) and big-endian (XDR, byte-order flag `0x00`) WKB payloads are supported. This applies to all geometry types rendered to raster tiles (Point, LineString, Polygon, and their Multi- variants).
```
[the remark this answers is not in the document]
Comment (mikemcdougall) [reply docs/user/specifications/ogc-api-tiles-coverage.md:28]: This is addressed by commit `3d648047`. The PR diff shows `TileRenderer.cs` now reads the byte-order flag at `wkb[0]` (line 89) and all coordinate/type reads use `ReadInt32`/`ReadDouble` helpers backed by `BinaryPrimitives.ReadInt32LittleEndian`/`BigEndian` and `ReadDoubleLittleEndian`/`BigEndian` (lines 254-268). Three new unit tests (`TileRendererWkbTests`) verify big-endian, little-endian, and byte-order parity.
